src/ui/src/js/controllers/admin_garden.js

```diff
@@ -53,15 +53,32 @@ export default function adminGardenController(
     loadGardens();
   };
 
-  EventService.addCallback('admin_system', (event) => {
+  EventService.addCallback('admin_garden', (event) => {
     switch (event.name) {
       case 'GARDEN_CREATED':
-        break;
+        $scope.data.push(event.payload);
+        $apply();
       case 'GARDEN_REMOVED':
-        break;
+        let new_data = []
+        for (var i = 0; i < $scope.data.length; i++) {
+            if ($scope.data[i].id != event.payload.id){
+                new_data.push($scope.data[i]);
+            }
+        }
+        $scope.data = new_data
+      case 'GARDEN_UPDATED':
+        for (var i = 0; i < $scope.data.length; i++) {
+            if ($scope.data[i].id == event.payload.id){
+                $scope.data[i] = event.payload;
+            }
+        }
     }
```

Comment:
Since you're modifying the contents of a scope data structure (rather than replacing it) add a `$scope.digest()` call at the end to update all the bindings.

Reply:
$scope.disgest() did not resolve. Everything seems to now update dynamically after modifying the object in $scope.

Reply:
Ehh, it should have been `$scope.$digest()`. But if things seem snappy without it then :shrug: